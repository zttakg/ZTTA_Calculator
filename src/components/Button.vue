<script>
export default {
  name: "Button",

  props: {
    type: {
      type: String,
      default: "button"
    },
    color: String,
    to: {
      type: String,
      default: null
    },
    size: {
      type: String,
      default: "medium"
    },
    theme: String,
    block: Boolean,
    disabled: Boolean,
    icon: Boolean,
    value: null
  },

  computed: {
    classes() {
      /* eslint-disable indent */
      const themes = this.theme
        ? this.theme
            .split(" ")
            .filter(t => t !== "")
            .map(t => `button_style_${t}`)
        : [];
      /* eslint-enable indent */

      const classes = [
        "button",
        {
          [`button_color_${this.color}`]: this.color,
          [`button_size_${this.size}`]: this.size,
          button_block: this.block,
          button_disabled: this.disabled,
          button_icon: this.icon
        },
        ...themes
      ];

      return classes;
    }
  },

  methods: {
    handleClick(event) {
      this.$emit("click", event);
    }
  },

  // eslint-disable-next-line no-unused-vars
  render(h) {
    const ButtonTag = this.to ? "a" : "button";
    const type = this.to ? null : this.type;

    return (
      <ButtonTag
        onClick={this.handleClick}
        class={this.classes}
        type={type}
        href={this.to}
      >
        {this.$slots.default}
      </ButtonTag>
    );
  }
};
</script>
