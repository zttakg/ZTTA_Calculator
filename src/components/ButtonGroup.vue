
<script>
export default {
  name: "ButtonGroup",

  props: {
    baseClassName: {
      type: String,
      default: "button-group"
    }
  },

  // eslint-disable-next-line no-unused-vars
  render(h) {
    return (
      <div class={this.baseClassName}>
        {this.$slots.default &&
          // eslint-disable-next-line arrow-parens
          this.$slots.default.map(vnode => {
            const clonedVnode = vnode;
            clonedVnode.data.class = {};
            clonedVnode.data.class["button-group__item"] = true;
            return clonedVnode;
          })}
      </div>
    );
  }
};
</script>
