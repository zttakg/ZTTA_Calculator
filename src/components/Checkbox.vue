<template>
  <label :class="classes" :for="id">
    <input
      class="form-control form-control_type_checkbox checkbox__control"
      type="checkbox"
      :id="id"
      :name="name"
      :value="value"
      :class="className"
      :required="required"
      :disabled="disabled"
      @change="onChange"
      :checked="state"
    />
    <span class="checkbox__box"></span>
    <span v-if="label || hasSlot()" class="checkbox__label">
      {{ label }}
      <slot></slot>
    </span>
  </label>
</template>

<script>
import uid from "@/helpers/generateId";
import SlotMixin from "@/mixins/slot";

export default {
  name: "Checkbox",

  mixins: [SlotMixin],

  model: {
    prop: "modelValue",
    event: "input"
  },

  props: {
    id: {
      type: String,
      default() {
        return `checkbox-id-${uid()}`;
      }
    },
    name: {
      type: String,
      default: null
    },
    value: {
      default: ""
    },
    modelValue: {
      default: undefined
    },
    className: {
      type: String,
      default: null
    },
    theme: {
      type: String,
      default: "default"
    },
    label: String,
    checked: {
      type: Boolean,
      default: false
    },
    required: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean,
      default: false
    },
    model: {}
  },

  computed: {
    classes() {
      const themes = this.theme
        .split(" ")
        .filter(t => t !== "")
        .map(t => `checkbox_style_${t}`);

      return ["checkbox", ...themes];
    },
    state() {
      if (this.modelValue === undefined) {
        return this.checked;
      }
      if (Array.isArray(this.modelValue)) {
        return this.modelValue.indexOf(this.value) > -1;
      }
      return !!this.modelValue;
    }
  },

  watch: {
    checked(newValue) {
      if (newValue !== this.state) {
        this.toggle();
      }
    }
  },

  mounted() {
    if (this.checked && !this.state) {
      this.toggle();
    }
  },

  methods: {
    onChange() {
      this.toggle();
    },
    toggle() {
      let value;
      if (Array.isArray(this.modelValue)) {
        value = this.modelValue.slice(0);
        if (this.state) {
          value.splice(value.indexOf(this.value), 1);
        } else {
          value.push(this.value);
        }
      } else {
        value = !this.state;
      }
      this.$emit("input", value);
    }
  }
};
</script>
