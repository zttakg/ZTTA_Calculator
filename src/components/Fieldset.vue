<script>
export default {
  name: "Fieldset",

  props: {
    legend: String
  },

  // eslint-disable-next-line no-unused-vars
  render(h) {
    return (
      <fieldset class="fieldset">
        {this.legend && <legend class="fieldset__legend">{this.legend}</legend>}
        {this.$slots.default &&
          this.$slots.default.map(child => (
            <div class="fieldset__item">{child}</div>
          ))}
      </fieldset>
    );
  }
};
</script>
