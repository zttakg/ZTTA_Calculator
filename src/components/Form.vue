<script>
export default {
  name: "Form",

  // eslint-disable-next-line no-unused-vars
  render(h) {
    return <form class="form">{this.$slots.default}</form>;
  }
};
</script>
