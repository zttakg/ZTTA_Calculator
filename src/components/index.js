export { default as Button } from "./Button";
export { default as Checkbox } from "./Checkbox";
export { default as Fieldset } from "./Fieldset";
export { default as Form } from "./Form";
export { default as FormItem } from "./FormItem";
export { default as Icon } from "./Icon";
export { default as MaterialIcon } from "./MaterialIcon";
export { default as Input } from "./Input";
export { default as Radio } from "./Radio";
export { default as Switcher } from "./Switcher";
export { default as Select } from "./Select";
export { default as ButtonGroup } from "./ButtonGroup";
export { default as InputGroup } from "./InputGroup";
export { default as Dialog } from "./Dialog";
export { default as Tooltip } from "./Tooltip";

// svg
export { default as SvgRoot } from "./SvgRoot";
export { default as SvgRectangle } from "./SvgRectangle";
export { default as SvgCircle } from "./SvgCircle";
export { default as SvgPolygon } from "./SvgPolygon";
export { default as SvgTriangle } from "./SvgTriangle";
export { default as SvgLine } from "./SvgLine";
export { default as SvgPath } from "./SvgPath";
export { default as SvgArrow } from "./SvgArrow";
export { default as SvgText } from "./SvgText";
export { default as DimensionLine } from "./DimensionLine";
