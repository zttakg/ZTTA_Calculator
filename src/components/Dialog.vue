<template>
  <div
    :class="classes"
    :id="name"
    role="dialog"
    aria-modal="true"
    :aria-labelledby="`${name}-dialog__title`"
  >
    <header v-if="(title || close)" class="dialog__header">
      <h2 class="dialog__title" :id="`${name}-dialog__title`">{{ title }}</h2>
      <Button v-if="close" @click="close" class="dialog__close">
        <Icon name="close" />
        <span class="visuallyhidden">Close this dialog window</span>
      </Button>
    </header>

    <main class="dialog__content">
      <slot></slot>
    </main>

    <footer class="dialog__footer">
      <slot name="footer"></slot>
    </footer>
  </div>
</template>

<script>
import { Modal } from "vuex-modal";
import { Button, Icon } from "@/components";

export default {
  name: "Dialog",

  components: {
    Button,
    Icon,
    Modal
  },

  props: {
    name: {
      type: String,
      default: "dialog"
    },
    theme: String,
    title: String,
    close: Function
  },

  computed: {
    classes() {
      /* eslint-disable indent */
      const themes = this.theme
        ? this.theme
            .split(" ")
            .filter(t => t !== "")
            .map(t => `dialog_theme_${t}`)
        : [];
      /* eslint-enable indent */

      const classes = ["dialog", ...themes];

      return classes;
    }
  }
};
</script>
