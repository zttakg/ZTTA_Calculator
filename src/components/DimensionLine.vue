<template>
  <g :data-name="`dimension-line dimension-line_${orientation}`">
    <SvgLine
      v-for="(line, index) in lines"
      :data-name="`dimension-line__${line.name}`"
      :key="index"
      :from="line.from"
      :to="line.to"
      :strokeColor="lineColor"
      stroke-linecap="square"
      shape-rendering="crispEdges"
      stroke-dasharray="3,3"
    />
    <SvgArrow
      :orientation="orientation"
      :width="arrowSize.width"
      :height="arrowSize.height"
      :transform="transform"
      :color="lineColor"
    />
    <SvgText
      data-name="dimension-line__text"
      :dx="text.position[0]"
      :dy="text.position[1]"
      :transform="`rotate(${text.rotate})`"
      text-anchor="middle"
      :style="textStyle"
    >
      {{ text.title }}
    </SvgText>
  </g>
</template>

<script>
import { SvgArrow, SvgLine, SvgText } from "@/components";
import {
  DIMENSION_LINE_OFFSET,
  DIMENSION_LINE_COLOR,
  DIMENSION_LINE_TEXT_STYLE
} from "@/constants";

export default {
  name: "DimensionLine",

  components: {
    SvgLine,
    SvgArrow,
    SvgText
  },

  props: {
    orientation: {
      type: String,
      default: "horizontal" // or "vertical"
    },
    width: {
      type: String,
      default: "300"
    },
    height: {
      type: String,
      default: "300"
    },
    diameter: {
      type: String,
      default: "300"
    },
    side1: {
      type: String,
      default: "300"
    },
    side2: {
      type: String,
      default: "300"
    },
    type: {
      type: String,
      default: "rectangle"
    },
    radius: {
      type: String,
      default: "0"
    }
  },

  data() {
    return {
      lineColor: DIMENSION_LINE_COLOR,
      textStyle: DIMENSION_LINE_TEXT_STYLE
    };
  },

  computed: {
    arrowSize() {
      switch (this.type) {
        case "rectangle":
          return {
            width: this.width,
            height: this.height
          };

        case "circle":
          return {
            width: this.diameter,
            height: this.diameter
          };

        case "triangle":
          return {
            width: this.side2,
            height: this.side1
          };

        default:
          return {
            width: 0,
            height: 0
          };
      }
    },
    lines() {
      const width = parseInt(this.width, 10);
      const height = parseInt(this.height, 10);
      const cornerRadius = parseInt(this.radius, 10);
      const diameter = parseInt(this.diameter, 10);
      const side1 = parseInt(this.side1, 10);
      const side2 = parseInt(this.side2, 10);
      const offset = DIMENSION_LINE_OFFSET;

      const lines = {
        rectangle: {
          horizontal: [
            {
              from: [1, height + 1 - cornerRadius],
              to: [1, height + offset + 1],
              name: "left"
            },
            {
              from: [width + 1, height + 1 - cornerRadius],
              to: [width + 1, height + offset + 1],
              name: "right"
            }
          ],
          vertical: [
            {
              from: [width + 1 - cornerRadius, 1],
              to: [width + offset + 1, 1],
              name: "top"
            },
            {
              from: [width + 1 - cornerRadius, height + 1],
              to: [width + offset, height + 1],
              name: "bottom"
            }
          ]
        },
        circle: {
          horizontal: [
            {
              from: [1, diameter / 2],
              to: [1, diameter + offset],
              name: "left"
            },
            {
              from: [diameter + 1, diameter / 2],
              to: [diameter + 1, diameter + offset],
              name: "right"
            }
          ],
          vertical: [
            {
              from: [diameter / 2, 1],
              to: [diameter + offset, 1],
              name: "top"
            },
            {
              from: [diameter / 2, diameter + 2],
              to: [diameter + offset, diameter + 1],
              name: "bottom"
            }
          ]
        },
        triangle: {
          horizontal: [
            {
              from: [1, side1 + 1],
              to: [1, side1 + offset + 1],
              name: "left"
            },
            {
              from: [side2, side1 + 1],
              to: [side2, side1 + offset + 1],
              name: "right"
            }
          ],
          vertical: [
            {
              from: [1, 1],
              to: [side2 + offset + 1, 1],
              name: "top"
            },
            {
              from: [side2, side1 + 1],
              to: [side2 + offset, side1 + 1],
              name: "bottom"
            }
          ]
        }
      };

      return lines[this.type][this.orientation];
    },
    text() {
      const shapeWidth = parseInt(this.width, 10);
      const shapeHeight = parseInt(this.height, 10);
      const diameter = parseInt(this.diameter, 10);
      const side1 = parseInt(this.side1, 10);
      const side2 = parseInt(this.side2, 10);
      const offset = DIMENSION_LINE_OFFSET;
      const TEXT_OFFSET = 15;

      function generateText(width, height = width) {
        return {
          horizontal: {
            title: `${width.toString()}\u00A0мм`,
            position: [width / 2, height + offset - TEXT_OFFSET],
            rotate: 0
          },
          vertical: {
            title: `${height.toString()}\u00A0мм`,
            position: [-height / 2, width + offset - TEXT_OFFSET],
            rotate: -90
          }
        };
      }

      const text = {
        rectangle: generateText(shapeWidth, shapeHeight),
        circle: generateText(diameter),
        triangle: generateText(side2, side1)
      };

      return text[this.type][this.orientation];
    },
    transform() {
      const offset = 8;
      return this.orientation === "horizontal"
        ? `translate(0, -${offset})`
        : `translate(-${offset}, 0)`;
    }
  }
};
</script>
