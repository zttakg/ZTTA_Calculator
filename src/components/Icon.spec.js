import { shallowMount } from "@vue/test-utils";
import { Icon } from "@/components";

jest.mock("@/constants", () => ({
  SVG_ICONS_SPRITE_PATH: "path/to/sprite.svg"
}));

describe("<Icon /> component", () => {
  it("should render correct without props", () => {
    const wrapper = shallowMount(Icon);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render all props", () => {
    const wrapper = shallowMount(Icon, {
      propsData: {
        name: "test",
        title: "Test icon"
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
