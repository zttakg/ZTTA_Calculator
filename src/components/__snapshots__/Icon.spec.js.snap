// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> component should render all props 1`] = `
<svg role="img" class="icon icon_test">
  <title>Test icon</title>
  <use xlink:href="path/to/sprite.svg#test"></use>
</svg>
`;

exports[`<Icon /> component should render correct without props 1`] = `undefined`;
